smart_open/smart_open_lib.py

```diff
@@ -754,11 +754,11 @@ def HttpOpenRead(parsed_uri, mode='r', **kwargs):
 
     response = HttpReadStream(url, **kwargs)
 
-    fname = url.split('/')[-1]
+    fname = urlsplit(url, allow_fragments=False).path.split('/')[-1]
```

Comment:
I was trying to stay consistent with the previous use of the urlsplit function 

https://github.com/RaRe-Technologies/smart_open/blob/master/smart_open/smart_open_lib.py#L246